truss/templates/server/common/logging.py

```diff
@@ -5,11 +5,44 @@
 
 LEVEL: int = logging.INFO
 
+
+class CustomJsonFormatter(jsonlogger.JsonFormatter):
+    """
+    A custom JsonFormatter to reformat the web server log entries.
+    """
+
+    def process_log_record(self, log_record):
+        message = log_record.get("message", "")
+
+        # Parse the necessary information from the message
+        parts = message.split(" ")
+        request_type = parts[2][1:]
+        endpoint = parts[3].split("%3A")[1]
+        status_code = parts[5]
```

Comment:
What happens in a log records doesn't have the right number of parts you expect? Does it make sense to do a try/except so that these IndexErrors don't result in crashes and just default to not changng the message in case of an issue?

Reply:
Is it possible to infer the supposed format of the log from some property on the [LogRecord](https://docs.python.org/3/library/logging.html#logrecord-attributes) itself rather than catching an error?

Reply:
Hmmm, doesn't look like it. All of the important information is passed into the `message`.